### PR TITLE
Add GitVersionHelper (take 2)

### PR DESCRIPTION
**Pull request by trondd**
Re-implemented with Newtonsoft.Json. Hopefully no trouble this time.

### CODE REVIEW

File: src/app/FakeLib/GitVersionHelper.fs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                                NuGetVersionV2 : string;
+                                NuGetVersion : string;
+                                CommitsSinceVersionSource : int;
+                                CommitsSinceVersionSourcePadded : string;
+                                CommitDate : string;
+                            }
+
+
+let GitVersion () =
+    let dirs = [ (environVar "ProgramData")  @@ "chocolatey" @@ "bin" ]
```
Comment (forki) [inline src/app/FakeLib/GitVersionHelper.fs]: is that the most common place?

Comment (baronfel) [inline src/app/FakeLib/GitVersionHelper.fs]: Gitversion does suggest using chocolatey to install the exe, but I think a more resilient way to find the path to the chocolatey install directory is using the environment variable `ChocolateyInstall`. Then we can add `bin` to that and get the install dir.

File: src/app/FakeLib/GitVersionHelper.fs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    let file = "GitVersion.exe"
+    let timespan =  TimeSpan.FromMinutes 1.
+    let usage = "To install GitVersion.exe, start PowerShell as Administrator and run choco install gitversion.portable -s https://chocolatey.org/api/v2"
```
Comment (baronfel) [inline src/app/FakeLib/GitVersionHelper.fs]: [This](http://gitversion.readthedocs.io/en/latest/usage/usage/) describes the common usages.  I'd expect that most people would get the exe from there, especially as the exe is the primary way to get support for build/CI servers as described [here](http://gitversion.readthedocs.io/en/latest/usage/usage/)

Comment (forki) [inline src/app/FakeLib/GitVersionHelper.fs]: I think we need to make that work on mono and windows.

Comment (baronfel) [inline src/app/FakeLib/GitVersionHelper.fs]: There's a [gitversion.commandline](https://www.nuget.org/packages/GitVersion.CommandLine) nuget package, so perhaps we should use the existing tools-path-finding that we do for other nuget-delivered exes? and then we've got conventions for executing those on mono already in FAKE.

**Comment by trondd** [comment]
Btw, this now works on Mac as well with the NuGet version of GitVersion

**Comment by forki** [comment]
great!

**Pull request merged**